lib/crawler.js

```diff
@@ -1096,17 +1154,19 @@ Crawler.prototype.fetchQueueItem = function(queueItem) {
 
         queueItem.fetched = true;
         queueItem.status = "timeout";
+        clientRequest.abort();
 
+        /**
+         * Fired when a request times out
+         * @event Crawler#fetchtimeout
+         * @param {QueueItem} queueItem The queue item for which the request timed out
+         * @param {Number} timeout      The delay in milliseconds after which the request timed out
+         */
         crawler.emit("fetchtimeout", queueItem, crawler.timeout);
-        clientRequest._crawlerHandled = true;
-        clientRequest.abort();
     });
 
-    clientRequest.on("error", function(errorData) {
-
-        // This event will be thrown if we manually aborted the request,
-        // but we don't want to do anything in that case.
-        if (clientRequest._crawlerHandled) {
+    clientRequest.on("error", function(error) {
+        if (clientRequest.aborted) {
```

Comment:
We can rely on this property set by node itself when a request is aborted rather than a manual one. It expresses intent more clearly as well